share/availability/discovery/discovery.go

```diff
@@ -273,12 +275,18 @@ func (d *Discovery) findPeers(ctx context.Context) {
 		return
 	}
 
+	ticker := time.NewTicker(findPeersStuckWarnDelay)
+	defer ticker.Stop()
 	var amount int
 	for {
+		ticker.Reset(findPeersStuckWarnDelay)
 		select {
 		case <-findCtx.Done():
 			log.Debugw("found enough peers", "amount", d.set.Size())
 			return
+		case <-ticker.C:
+			log.Error("wasn't able to find new peers for long time")
+			continue
```

Comment:
Does this happen in practise?

Reply:
Very unlikely, as you probably expected. Bit It could in certain conditions:
- if there are not enough good peers in network
- if discovery reached internal discovery limit and we still didn't found enough peers
- if target limit set high (like >100) it will happen often (not our real case)

Ideally we should not see it, but if we do, it means discovery is unable to give new peers to connect. I want this case to be explicitly logged, so if it happen we get reports and investigate.